### PR TITLE
docs(NODE-3314): Changed '.equal()' to '.strictEqual()' in examples

### PR DESCRIPTION
**Pull request by lMartin3**
asserts.equal() is deprecated, changed it to assert.strictEqual().

## Description

**What changed?**
README.md - Changed the examples so they used asserts.strictEqual() instead of asserts.equal() (which is deprecated).

**Are there any files to ignore?**
No

### CODE REVIEW

**Review by nbbeeken — APPROVED** [review]
Seems like a good idea to modernize the examples, thanks for your contribution!

**Review by dariakp — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -226,9 +226,9 @@ const insertDocuments = function(db, callback) {
   const collection = db.collection('documents');
   // Insert some documents
   collection.insertMany([{ a: 1 }, { a: 2 }, { a: 3 }], function(err, result) {
-    assert.equal(err, null);
-    assert.equal(3, result.result.n);
-    assert.equal(3, result.ops.length);
+    assert.strictEqual(err, null);
```
Comment (emadum) [inline README.md:229]:
```suggestion
    assert.ifError(err);
```

I think we could go a bit further and update the error checks to use `assert.ifError` here, which seems like the most idiomatic approach.

**Review by emadum — CHANGES_REQUESTED** [review]
Thanks for this PR! Just one small suggestion.

**Comment by wa-Nadoo** [comment]
IMHO the better approach is to enforce strict assertion mode for all methods. Just replace `const assert = require('assert')` with  `const assert = require('assert').strict` or  `const assert = require('assert/strict')`. In strict assertion mode all non-strict methods behave like strict methods. More info is available here: https://nodejs.org/api/assert.html#assert_strict_assertion_mode.

**Comment by nbbeeken** [comment]
Closing due to inactivity and target branch needs updating, 3.7 is now the current 3.x branch

**Pull request closed without merging**